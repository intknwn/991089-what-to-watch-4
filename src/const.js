export const STATIC_RESOURSES_URL = `https://4.react.pages.academy`;

export const MAX_GENRES_COUNTER = 9;
export const MAX_MORE_LIKE_THIS_MOVIES_COUNTER = 4;
export const MOVIES_PER_PAGE = 8;

export const MOVIES_LIST_PLAYER_CONFIG = {
  autoPlay: true,
  muted: true,
  loop: true,
};

export const MOVIE_PAGE_PLAYER_CONFIG = {
  autoPlay: false,
  muted: false,
  loop: false,
};

export const Tab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const Genre = {
  ALL_GENRES: `All genres`,
};

export const AuthStatus = {
  AUTH: `AUTH`,
  NOT_AUTH: `NOT_AUTH`,
};

export const HttpCode = {
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  INTERNAL_ERROR: 500,
  SERVICE_UNAVAILABLE: 503,
};

export const ReviewForm = {
  MIN_REVIEW_LENGTH: 50,
  MAX_REVIEW_LENGTH: 400,
  MAX_RATE: 5,
};

export const AppRoute = {
  MAIN: `/`,
  SIGN_IN: `/sign-in`,
  MY_LIST: `/mylist`,
  REVIEW: `/review`,
  PLAYER: `/player`,
  MOVIE: `/movies`,
};

export const ClassName = {
  MOVIE_PAGE_HEADER: `page-header movie-card__head`,
  USER_PAGE_HEADER: `page-header user-page__head`,
  LOGO_LINK: `logo__link`,
  LOGO_LINK_LIGHT: `logo__link logo__link--light`,
};
