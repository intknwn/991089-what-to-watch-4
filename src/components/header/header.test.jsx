import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Header from '../header/header.jsx';
import {ClassName} from '../../const.js';

const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarSrc: `/img/1.png`,
};

it(`Header component should render correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <Header
          className={ClassName.MOVIE_PAGE_HEADER}
          isAuthorized={true}
          isEmpty={false}
          user={user}
        />
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
