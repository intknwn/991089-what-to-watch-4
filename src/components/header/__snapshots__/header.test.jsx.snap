// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="Oliver.conner@gmail.com"
          height={63}
          src="https://4.react.pages.academy/img/1.png"
          width={63}
        />
      </a>
    </div>
  </div>
</header>
`;
